### PR TITLE
feat(provider/appengine): Container deploy w/ free url or artifact

### PR DESCRIPTION
**Pull request by sbwsg**
appengine_pipeline_artifact

### CODE REVIEW

File: app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.controller.ts [hunk 1]
```diff
@@ -53,6 +53,9 @@ class AppengineServerGroupBasicSettingsCtrl implements IController {
     }
 
     const artifact = copy(expected.matchArtifact);
+    if (artifact == null) {
+      return '';
+    }
     return Object.keys(artifact)
```
Comment (lwander) [inline app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.controller.ts]: I think there is a filter registered for this, something like summarize-expected-artifacts

**Comment by sbwsg** [comment]
:+1: screenshot added.

File: app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.controller.ts (hunk 1 above)
Comment (sbwsg) [reply app/scripts/modules/appengine/serverGroup/configure/wizard/basicSettings.controller.ts]: I'll take another look here.  I saw exceptions in this function around fetching keys for null/undefined.  They occurred after deleting an Expected Artifact from a Configuration stage that was in use in a Deploy stage.

**Comment by lwander** [comment]
Looking good!

What happens when GCS is selected, and they want to deploy a docker container from an artifact?

**Comment by sbwsg** [comment]
Fair question!  Currently that isn't supported.  I'll update this PR once work is done to support separate `app.yaml` and container artifacts in clouddriver.

**Comment by sbwsg** [comment]
@lwander I've removed the `if (artifact == null)` checks and will follow up on multiple-artifact appengine deploys in a new PR.  Otherwise ok to merge?

**Comment by lwander** [comment]
SGTM!

**Pull request merged**